pants-plugins/internal_plugins/test_lockfile_fixtures/lockfile_fixture.py

```diff
@@ -40,7 +34,13 @@ def __init__(
                 raise TypeError(
                     f"Unsupported type `{type(requirement)}` for JVM coordinate. Expected `Coordinate` or `str`."
                 )
-        self.requirements = tuple(coordinates)
+
+        object.__setattr__(
+            self,
+            "lockfile_rel_path",
```

Comment:
I fear typos in these, but I guess we have to pick our poison...

Reply:
Yeah I do despise stringly-typed languages 😕

Reply:
should be possible to have a lint/typing plugin to help check these strings for being valid attributes of `self`.

Reply:
You have your wish. 

I just added a `pytest` hook that monkeypatches `dataclass` and checks in `__init__` that our initialization matches the declared fields. And then I added tests for that, to make sure it is working :smile:

Reply:
Whoa!! Sophisticated!